### ISSUE TITLE
example code sometimes retains vertical bars

### ISSUE THREAD

**Issue by wkeese** [csnover/js-doc-parse#12]
Sometimes  the vertical bars end up in the XML output file

For example, when parsing dojo/_base/window.js, the definition for doc:

``` javascript
/*=====
doc: {
    // summary:
    //      Alias for the current document. 'doc' can be modified
    //      for temporary context shifting. See also withDoc().
    // description:
    //      Use this rather than referring to 'window.document' to ensure your code runs
    //      correctly in managed contexts.
    // example:
    //  |   n.appendChild(dojo.doc.createElement('div'));
},
=====*/
```

It turns into:

``` xml
<property name="doc" scope="normal" type="object" from="dojo/_base/window">
<summary>
Alias for the current document. 'doc' can be modified for temporary context shifting. See also withDoc().
</summary>
<description>
<p>Use this rather than referring to &#39;window.document&#39; to ensure your code runs correctly in managed contexts.</p>
</description>
<examples>
<example>
<p>| n.appendChild(dojo.doc.createElement(&#39;div&#39;));</p>
</example>
</examples>
</property>
```

Also some output from the Deferred modules:

``` xml
<example>
<p>Deferred objects are often used when making code asynchronous. It may be easiest to write functions in a synchronous 
manner and then split code using a deferred to trigger a response to a long-lived operation. For example, instead of
 register a callback function to denote when a rendering operation completes, the function can simply return a deferred: </p>
 <p></p> <pre><code> // callback style: function renderLotsOfData(data, callback){ var success = false try{ for(var x in data){ 
renderDataitem(data[x]); } success = true; }catch(e){ } if(callback){ callback(success); } }</code></pre>
 <p></p> 
<p>| // using callback style | renderLotsOfData(someDataObj, function(success){
 | // handles success or failure | if(!success){ | promptUserToRecover(); | } | });
 | // NOTE: no way to add another callback here!!</p>
</example>
```

cc @ttrenka

**Comment by csnover** [comment]
Fixed in d79ccce.

**Comment by csnover** [comment]
fwiw i consider this a bug with the docblocks which are putting the pipes marking them as examples wherever they want instead of flush with the property. there should be more consistency defined here.
